### ISSUE TITLE
Several storage types, keywords and primitives missing.

### ISSUE THREAD

**Issue by ysgard** [amehat/atom-language-d#2]
_Storage_: **in out inout immutable pure impure nothrow const auto cast export shared ref @property**
_Type/OO identification_: **typeid typeof this override virtual interface**
_Primitives_: **string ubyte ushort uint ulong real wchar dchar wstring dstring enum union struct**
_Testing_:  **unittest debug enforce assert**
_Statements/functions_: **foreach with asm mixin template scope** 
_Modules_: **module package version deprecated @safe @system @trusted**

assert and some of the other keywords loses its keyword highlighting and adopts function highlighting as soon as you use parentheses, it should retain its keyword highlighting as they are base constructs of the language.

**Comment by amehat** [comment]
Indeed, the basic idea was to prompt a support language D Atom , which is why I used Java as reference. It is now necessary to develop it so that it fully corresponds to D.

**Comment by jzhu98** [comment]
As per the [spec](http://dlang.org/lex.html), the keywords **impure**, **virtual** don't exist in D.

**Comment by jzhu98** [comment]
The state of implementation so far:

**Keywords**
- [ ] in
- [ ] out
- [x] inout
- [x] immutable
- [x] pure
- [x] nothrow
- [x] const
- [x] auto
- [x] cast
- [x] export
- [x] shared
- [x] ref
- [x] @property @safe @system @trusted
**Type/OO identification:**
- [x] typeid
- [x] typeof
- [x] this
- [x] override
- [x] interface
- [x] **Primitives:** string ubyte ushort int ulong real wchar dchar wstring dstring enum union struct

**Testing:**
- [x] unittest
- [x] debug
- [ ] enforce
- [x] assert

**Statements/functions:**
- [x] foreach
- [x] with
- [x] asm
- [x] mixin
- [x] template
- [x] scope

**Modules:**
- [x] module
- [x] package
- [x] version
- [x] deprecated
- [x] @safe @system @trusted

**Comment by MineyMe** [comment]
Another missing keyword is **synchronized**, which als messes up syntax highlighting right now:
```
synchronized
{
   // Everything in this block will be synchronized
}

// Without braces...
synchronized if (some_condition)
   do_something();

// As modifier for variables...
synchronized this.some_variable = true;
```

**Comment by luke5542** [comment]
I don't see the `body` keyword in that list either, and I have noticed some highlighting errors related to its use, or lack of use to be more accurate.

For example, in an interface where you want to add contracting to a method, leaving out the body section, as you are supposed to, leaves the rest of the file unhighlighted.
```D
interface MyInterface {
    double fun(double arg1)
    in {
        assert(some_pre_condition);
    }
    out {
        assert(some_post_condition);
    }
    // Normal functions would have a body {...} here as well,
    // but right now the rest of the file goes unhighlighted.
}
```
